File: src/main/java/leetcode/two_some_2/TwoSum2.java
package leetcode.two_some_2;

public class TwoSum2 {

    public static Integer[] twoSum(Integer[] aceOrderedItems, Integer target) {
        int left = 0;
        int right = aceOrderedItems.length - 1;
        while (left < right) {
            int value = aceOrderedItems[left] + aceOrderedItems[right];
            if (value > target)
                right--;
            else if (value < target)
                left++;
            else
                return new Integer[]{left, right};

        }
        return null;
    }

}

File: src/main/java/leetcode/two_some_2/TwoSum2.kt
package leetcode.two_some_2

fun twoSum(aceOrderedItems: Array<Int>, target: Int): Array<Int> {
    var left = 0
    var right: Int = aceOrderedItems.size - 1
    while (left < right) {
        val value: Int = aceOrderedItems[left] + aceOrderedItems[right]
        if (value > target) right--
        else if (value < target) left++
        else return arrayOf(left, right)
    }
    return emptyArray()
}